stable/vpa/templates/admission-controller-service.yaml

```diff
@@ -9,5 +9,17 @@ spec:
       targetPort: 8000
   selector:
     app.kubernetes.io/component: admission-controller
-    {{- include "vpa.labels" . | nindent 4 }}
+    {{- include "vpa.selectorLabels" . | nindent 4 }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: vpa-webhook
+spec:
+  ports:
+    - port: 443
+      targetPort: 8000
+  selector:
+    app.kubernetes.io/component: admission-controller
+    {{- include "vpa.selectorLabels" . | nindent 4 }}
```

Comment:
Perhaps I'm missing something, but this looks like the exact same service as above, but with a different name. Can you elaborate on why this is needed?

Reply:
The admission-controller pod creates a webhook, in this webhook is a service defined:

```kind: MutatingWebhookConfiguration
metadata:
  creationTimestamp: "2020-09-28T15:27:04Z"
  generation: 1
  name: vpa-webhook-config
  resourceVersion: "141705845"
  selfLink: /apis/admissionregistration.k8s.io/v1/mutatingwebhookconfigurations/vpa-webhook-config
  uid: 133e4c3b-0214-4b38-933b-69d60e290779
webhooks:
- admissionReviewVersions:
  - v1beta1
      name: **vpa-webhook**
      namespace: cpt
      port: 443
```

See also here below:
https://github.com/kubernetes/autoscaler/blob/8cd8009947a06949b9cd12c5a0d27f617c03e155/vertical-pod-autoscaler/deploy/admission-controller-deployment.yaml

Reply:
I get that, but your PR creates two duplicate services to the same port on the same dpeloyment.

Even the example you posted only shows a single service. Can you explain why we need  two of them?

Reply:
I think it's for metrics?
There also both in here
https://github.com/kubernetes/autoscaler/tree/8cd8009947a06949b9cd12c5a0d27f617c03e155/vertical-pod-autoscaler/deploy

Reply:
So remove the other one?

Reply:
I think just remove the duplicate one (the one you added), leave the first one, and make this PR just about fixing the label selector for the service.

Reply:
Ok, but i still changed the naming as it won't work with the old name (referencing a non existing service in the webhook created)